pkg/scheduler/preemption/preemption_test.go

```diff
@@ -2796,3 +2813,83 @@ func TestPreemptionMessage(t *testing.T) {
 		}
 	}
 }
+
+func hierarchyManagerComparer(x, y hierarchy.Manager[*cache.ClusterQueueSnapshot, *cache.CohortSnapshot]) bool {
+	if !clusterQueuesSnapshotsComparer(x.ClusterQueues(), y.ClusterQueues()) {
+		return false
+	}
+	if !cohortSnapshotsComparer(x.Cohorts(), y.Cohorts()) {
+		return false
+	}
+	return true
+}
+
+func clusterQueuesSnapshotsComparer(x, y map[kueue.ClusterQueueReference]*cache.ClusterQueueSnapshot) bool {
+	if len(x) != len(y) {
+		return false
+	}
+	for name, xSnapshot := range x {
+		ySnapshot, ok := y[name]
+		if !ok {
+			return false
+		}
+		if !clusterQueueSnapshotComparer(xSnapshot, ySnapshot) {
+			return false
+		}
+	}
+	return true
+}
+
+func clusterQueueSnapshotComparer(x, y *cache.ClusterQueueSnapshot) bool {
+	// need to evaluate at every level of the hierarchy
+	opt := append(SnapCmpOpts(),
+		cmpopts.IgnoreFields(cache.ClusterQueueSnapshot{}, "ClusterQueue", "tasOnly"),
+		cmpopts.IgnoreFields(hierarchy.ClusterQueue[*cache.CohortSnapshot]{}, "cohort"),
+	)
+
+	diff := cmp.Diff(x, y, opt...)
+	if diff != "" {
+		return false
+	}
+	if x == nil {
+		return true
+	}
+	if !cohortSnapshotComparer(x.ClusterQueue.Parent(), y.ClusterQueue.Parent()) {
+		return false
+	}
+	return true
+}
+
+func cohortSnapshotsComparer(x, y map[kueue.CohortReference]*cache.CohortSnapshot) bool {
+	if len(x) != len(y) {
+		return false
+	}
+	for name, xSnapshot := range x {
+		ySnapshot, ok := y[name]
+		if !ok {
+			return false
+		}
+		if !cohortSnapshotComparer(xSnapshot, ySnapshot) {
+			return false
+		}
+	}
+	return true
+}
+
+func cohortSnapshotComparer(x, y *cache.CohortSnapshot) bool {
+	opt := append(SnapCmpOpts(),
```

Comment:
Here the same. Don't use this SnapCmpOpts() here. If I correct understood you need just:

```
cmp.Comparer(func(x, y cache.ResourceNode) bool {
	return cmp.Equal(x.Quotas, y.Quotas) && cmp.Equal(x.SubtreeQuota, y.SubtreeQuota)
})
```